contracts/ExitFormat.sol

```diff
@@ -23,14 +17,30 @@ library ExitFormat {
 
     // A SingleAssetExit specifies
     // * an asset address (0 implies the native asset of the chain: on mainnet, this is ETH)
-    // * custom metadata (optional field, can be zero bytes). This might specify how to transfer this particular asset (e.g. target an "ERC20.transfer"' method)
+    // * custom tokenMetadata
+    //   containing
+    //   - TokenType enum value
+    //   - metadata for that token
     // * an allocations array
     struct SingleAssetExit {
         address asset;
-        bytes metadata;
+        TokenMetadata tokenMetadata;
```

Comment:
Note: this is a breaking change. We could maintain backwards compatibility by encoding `TokenMetadata` into bytes. This would cost some more gas.

Reply:
This is much cleaner and more gas efficient for sure. It is just if a breaking change is viable at this stage. Maybe better to do it sooner rather than later?

Reply:
@willemolding we are comfortable with a breaking change. 

TODO: We will want to handle it by increasing the first nonzero portion of the package [semver](https://semver.npmjs.com/) which should prevent the breaking change reaching downstream packages until they explicitly upgrade.

If you like this approach, why not merge this into your PR, and then we can get a final review on #69?

Reply:
Will do!